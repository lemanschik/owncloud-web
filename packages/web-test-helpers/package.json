{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
