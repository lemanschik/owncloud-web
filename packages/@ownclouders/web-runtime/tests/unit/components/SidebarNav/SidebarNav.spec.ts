import SidebarNav from 'web-runtime/src/components/SidebarNav/SidebarNav.vue'
import sidebarNavItemFixtures from '../../../__fixtures__/sidebarNavItems'
import { createStore, defaultPlugins, defaultStoreMockOptions, mount } from 'web-test-helpers'

jest.mock('uuid', () => ({
  v4: () => {
    return '00000000-0000-0000-0000-000000000000'
  }
}))

const slots = {
  bottom: '<span class="footer">Footer</span>'
}

describe('OcSidebarNav', () => {
  it('displays a bottom slot if given', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.findAll('.footer').length).toBe(1)
  })
  it('renders navItems into a list', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('toggles into closed state upon button click', async () => {
    const { wrapper, storeOptions } = getWrapper()
    await wrapper.find('.toggle-sidebar-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(storeOptions.actions.closeNavigation).toHaveBeenCalled()
  })
  it('toggles back into open state upon button click', async () => {
    const { wrapper, storeOptions } = getWrapper()
    storeOptions.state.navigation.closed = true
    await wrapper.find('.toggle-sidebar-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(storeOptions.actions.openNavigation).toHaveBeenCalled()
  })
  it('initially sets the highlighter to the active nav item', async () => {
    const { wrapper } = getWrapper()
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.highlighterAttrs).toEqual({
      style: {
        transform: 'translateY(0px)',
        'transition-duration': '0.2s'
      }
    })
  })
})

function getWrapper() {
  const storeOptions = defaultStoreMockOptions
  const store = createStore(storeOptions)
  return {
    storeOptions,
    wrapper: mount(SidebarNav, {
      slots,
      props: {
        navItems: sidebarNavItemFixtures
      },
      global: {
        renderStubDefaultSlot: true,
        plugins: [...defaultPlugins(), store],
        stubs: { SidebarNavItem: true }
      }
    })
  }
}
