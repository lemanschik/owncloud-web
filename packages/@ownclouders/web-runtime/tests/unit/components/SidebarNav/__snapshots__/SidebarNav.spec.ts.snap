// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcSidebarNav renders navItems into a list 1`] = `
<div class="oc-app-navigation-expanded" id="web-nav-sidebar">
  <button aria-label="Toggle sidebar" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw toggle-sidebar-button-expanded oc-pr-s toggle-sidebar-button oc-pb-s oc-pt-m" type="button">
    <!-- @slot Content of the button -->
    <span class="oc-icon oc-icon-l oc-icon-passive">
      <!---->
    </span>
  </button>
  <nav aria-label="Sidebar navigation menu" class="oc-sidebar-nav oc-mb-m oc-mt-s oc-px-xs">
    <ul class="oc-list oc-my-rm oc-mx-rm">
      <div class="oc-ml-s oc-background-primary-gradient" id="nav-highlighter"></div>
      <sidebar-nav-item-stub active="true" collapsed="false" filltype="fill" icon="folder" index="00000000000000000000000000000000" name="Personal" target="[object Object]"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub active="false" collapsed="false" filltype="fill" icon="share-forward" index="00000000000000000000000000000000" name="Shares" target="[object Object]"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub active="false" collapsed="false" filltype="fill" icon="delete" index="00000000000000000000000000000000" name="Deleted files" target="[object Object]"></sidebar-nav-item-stub>
    </ul>
  </nav> <!-- @slot bottom content of the sidebar -->
  <span class="footer">Footer</span>
</div>
`;
